src/Symfony/Component/Routing/Tests/Annotation/RouteTest.php

```diff
@@ -12,35 +12,56 @@
 namespace Symfony\Component\Routing\Tests\Annotation;
 
 use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\PhpUnit\ExpectDeprecationTrait;
 use Symfony\Component\Routing\Annotation\Route;
 
 class RouteTest extends TestCase
 {
+    use ExpectDeprecationTrait;
+
+    /**
+     * @group legacy
+     */
     public function testInvalidRouteParameter()
     {
         $this->expectException(\BadMethodCallException::class);
         new Route(['foo' => 'bar']);
     }
 
+    /**
+     * @group legacy
+     */
     public function testTryingToSetLocalesDirectly()
     {
         $this->expectException(\BadMethodCallException::class);
         new Route(['locales' => ['nl' => 'bar']]);
     }
 
     /**
+     * @requires PHP 8
      * @dataProvider getValidParameters
      */
-    public function testRouteParameters($parameter, $value, $getter)
+    public function testRouteParameters(string $parameter, $value, string $getter)
+    {
+        $route = new Route(...[$parameter => $value]);
+        $this->assertEquals($route->$getter(), $value);
+    }
+
+    /**
+     * @group legacy
+     * @dataProvider getValidParametersLegacy
+     */
+    public function testRouteParametersLegacy(string $parameter, $value, string $getter)
```

Comment:
I think we always put the legacy word first, as in `testLegacyRouteParameters` or `provideLegacyValidParameters`

Reply:
Okay.